std/math/index.zig

```diff
@@ -242,6 +242,82 @@ test "math.max" {
     assert(max(i32(-1), i32(2)) == 2);
 }
 
+/// This function returns the minimum value of the integer type T.
+/// The result is a compile time constant.
+/// Tested in test/cases/misc.zig
+pub fn minValue(comptime T: type) @typeOf(42) {
+    comptime {
+        return switch (T) {
+            u1, u2, u3, u4, u5, u6, u7,
+            u8, u16, u32, u64, u128 => 0,
+            i1 => -1,
+            i2 => -2,
+            i3 => -4,
+            i4 => -8,
+            i5 => -16,
+            i6 => -32,
+            i7 => -64,
+            i8 => -128,
+            i16 => -32768,
+            i32 => -2147483648,
+            i64 => -9223372036854775808,
+            i128 => -170141183460469231731687303715884105728,
+            else => {
+                //Calculate for Integers that we do not have cached
+                return switch (@typeId(T)) {
+                  // - (1 << (T.bit_count - 1))
+                  TypeId.Int, TypeId.ComptimeInt => if (T.is_signed) -(1 << (T.bit_count - 1)) else 0,
+                  else => @compileError("minValue not implemented for " ++ @typeName(T)),
```

Comment:
Should this function also find the max value of an enum?
Use case: User made enum arrays

Reply:
I'm thinking it should probably just be `std.maxInt` and `std.minInt`. max/min values of enums can be separate functions.

Reply:
Okay, I will change the function name.